test/schema_test.cc

```diff
@@ -81,3 +81,443 @@ TEST(SchemaTest, Equality) {
   ASSERT_EQ(schema1, schema5);
   ASSERT_EQ(schema5, schema1);
 }
+
+class BasicShortNameTest : public ::testing::Test {
+ protected:
+  void SetUp() override {
+    field1_ = std::make_unique<iceberg::SchemaField>(1, "Foo", iceberg::int32(), true);
+    field2_ = std::make_unique<iceberg::SchemaField>(2, "Bar", iceberg::string(), true);
+    field3_ = std::make_unique<iceberg::SchemaField>(3, "Foobar", iceberg::int32(), true);
+
+    auto structtype = iceberg::StructType(
+        std::vector<iceberg::SchemaField>{*field1_, *field2_, *field3_});
+
+    auto listype = iceberg::ListType(iceberg::SchemaField::MakeRequired(
+        4, "element", std::make_shared<iceberg::StructType>(structtype)));
+
+    auto maptype =
+        iceberg::MapType(iceberg::SchemaField::MakeRequired(5, "key", iceberg::int32()),
+                         iceberg::SchemaField::MakeRequired(
+                             6, "value", std::make_shared<iceberg::ListType>(listype)));
+
+    field4_ = std::make_unique<iceberg::SchemaField>(
+        4, "element", std::make_shared<iceberg::StructType>(structtype), false);
+    field5_ = std::make_unique<iceberg::SchemaField>(5, "key", iceberg::int32(), false);
```

Comment:
Can we move `field4_` and `field5_` initialization before line 98? Then we can simply write it as 

```cpp
    auto maptype = std::make_shared<iceberg::MapType>(*field4_, *field5_);
``` 

The point is avoid creating duplicate objects.